src/gt4py/eve/visitors.py

```diff
@@ -158,13 +173,7 @@ def generic_visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
                     if (new_child := self.visit(child, **kwargs)) is not NOTHING
                 }
             )
-            if self.PRESERVED_ANNEX_ATTRS and (old_annex := getattr(node, "__node_annex__", None)):
-                # access to `new_node.annex` implicitly creates the `__node_annex__` attribute in the property getter
-                new_annex_dict = new_node.annex.__dict__
-                for key in self.PRESERVED_ANNEX_ATTRS:
-                    if (value := getattr(old_annex, key, NOTHING)) is not NOTHING:
-                        assert key not in new_annex_dict
-                        new_annex_dict[key] = value
+            _preserve_annex(node, new_node, self.PRESERVED_ANNEX_ATTRS)
```

Comment:
What do you think about leaving the condition outside of the function? This is probably the hottest loop in the toolchain so I'd rather try to avoid a unnecessary function call if possible. Additionally, it would also make the function name fully correct :smile: 
```suggestion
            if preserved_annex_attrs and hasattr(node, "__node_annex__"):
                _preserve_annex(node, new_node, self.PRESERVED_ANNEX_ATTRS)
```

Reply:
I thought about this, but went for not duplicating the condition. Makes sense to me though, so I reverted that.